Platform/Interfaces/iadc.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * STM32 F4 ADC ABSTRACT INTERFACE
+ * Author:   William Kim
+ *
+ * Email:    wkim@utdallas.edu
+ *
+ * (c) 2024 Dallas Formula Racing - Embedded Firmware Team
+ * Formula SAE International Collegiate Chapter
+ * GPL-3.0 License
+ */
+
+#ifndef IADC_H
+#define IADC_H
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+namespace platform
+{
+
+    class IADC
+    {
+    public:
+        virtual ~IADC() = 0;
+
+        //adc operations
+        virtual uint16_t convert() const = 0;
+        virtual void writeToBuffer() = 0;
+
+    private:
+
+    protected:
+    std::shared_ptr<std::vector<uint16_t>> adc_buf; //container for adc read
+    size_t buffer_size; //unsigned
```

Comment:
I think that if we want to control the maximum size of the buffer, maybe it'd be better to use a normal array,  that is  unless the helper functions of the std::vector class are too nice to pass up on

Reply:
i think vector class member functions such as pushback would be very useful here, and any other member function defined would just be extra useful utility not defined in array